core/internal/common.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
```

Comment:
would `protoutil` be a slightly better name?

Reply:
'internal' is a special package name (I have just discovered this).
The exports from it are visible to sibling packages but not anywhere else.
ht/ Ismail for showing me this.

Reply:
But happy to change if you prefer.  Note the thing about 'util' in package names that I mentioned yesterday, the Go folk tend to steer people away from using words like that.

Reply:
`util` is indeed a poor name. It seemed like it contained functions that were related to the proto types.  Let's leave the package name as `internal`.  Would you mind changing the file name to something like `protohelpers.go`?

Reply:
Done.